internal/expr/parser.go

```diff
@@ -922,16 +930,36 @@ func (p *Parser) parseColumnsInsertExpr() (expression, bool, error) {
 	}
 	p.skipBlanks()
 
-	parenCol := p.colNum()
-	parenLine := p.lineNum
+	if sources, ok, err := p.parseColumnsInsertValues(); err != nil {
+		cp.restore()
+		return nil, false, err
+	} else if ok {
+		return &columnsInsertExpr{columns: columns, sources: sources, raw: p.input[cp.pos:p.pos]}, true, nil
+	}
+
+	if sources, ok, err := p.parseBasicInsertValues(); err != nil {
+		cp.restore()
+		return nil, false, err
+	} else if ok {
+		return &basicInsertExpr{columns: columns, sources: sources, raw: p.input[cp.pos:p.pos]}, true, nil
+	}
+
+	cp.restore()
+	return nil, false, nil
+}
+
+// parseColumnsInsertValues parses insert expressions that are value on the
+// right hand side of a columns insert expression.
```

Comment:
Can you state what the expression is instead of `columns insert expression`. I find that confusing as the outer parse function is named the same and there is no clear indication what the difference is.